scripts/run-live-build.sh

```diff
@@ -61,84 +62,38 @@ else
 	cd "$TOP/live-build/variants/$1"
 fi
 
-function get_s3_uri() {
-
-	local def_bucket="snapshot-de-images"
-	local jenkinsid="jenkins-ops"
-
-	local prefix="$1"
-	local latest="s3://$def_bucket/builds/$jenkinsid/$2"
-	local uri
-
-	if [[ -z "$prefix" ]]; then
-		#
-		# prefix is empty, so get latest version
-		#
-		aws s3 cp --quiet "$latest" .
-		prefix=$(cat latest)
-		uri="s3://$def_bucket/$prefix"
-		rm -f latest
-	elif [[ "$prefix" == s3* ]]; then
-		#
-		# prefix was set to full s3 URI: "s3://<bucket>/<prefix>"
-		#
-		uri="$prefix"
-	else
-		#
-		# We assume prefix is inside default bucket
-		#
-		uri="s3://$def_bucket/$prefix"
-	fi
+#
+# The ancillary repository contains all of the first-party Delphix
+# packages that are required for live-build to operate properly.
+#
 
-	if aws s3 ls "$uri" >/dev/null; then
-		echo "$uri"
-	else
-		echo "$uri not found." 1>&2
+aptly serve -config="$TOP/ancillary-repository/aptly.config" &
+APTLY_SERVE_PID=$!
+
+#
+# We need to wait for the Aptly server to be ready before we proceed;
+# this can take a few seconds, so we retry until it succeeds.
+#
+set +o errexit
+attempts=0
+while ! curl --output /dev/null --silent --head --fail \
+	"http://localhost:8080/dists/bionic/Release"; do
+	((attempts++))
+	if [[ $attempts -gt 30 ]]; then
+		echo "Timed out waiting for ancillary repository." 1>&2
+		kill $APTLY_SERVE_PID
```

Comment:
Should we do a kill -9?

Reply:
We could, but since this process will be running in the docker container, even if the process were to linger, it will be killed when the docker container exits.. so, I opted not to use it, in favor of letting the process handle the signal, and more gracefully exit.

If you'd rather use kill -9, I think that'll work fine, and I'm OK with that.